observer-errors/README.md

```diff
@@ -0,0 +1,130 @@
+## Explainer
+
+- __Authors__: [Leo Balter](@leobalter), [Greg Whitworth](@gregwhitworth)
+- __Expected graduation venue__: WHATWG
+- __Status__: Early Draft
+
+The `*Observer` APIs operate with callback functions that are triggered after observation of many events. There isn't much that can be done today in these APIs to capture and handle errors that happen either on the respective observations or abrupt completions that happen in the callback functions. These exceptions might end up leaking up to the Window's error event, potentially causing an application random hazards if they are not properly handled.
+
+Note: While the problem denoted above is experienced by any observer (Mutation, Performance, etc.), this explainer will be using the `ResizeObserver` for examples in order to scope the scenarios for ease of reader consumption.
+
+### Expected Exceptions in the Observer API
+
+There are at least three kinds of errors to be considered in the Observer model:
+
+#### Configuration
+
+Those are the exceptions thrown when you're setting up a new instance or setting elements to be observed, e.g.: on the immediate call for `o.observe(elem)`. Those are immediate errors and can be be captured today.
+
+Another example would be using an invalid argument for the `ResizeObserver`'s callback function. Users can handle this exception today, there is nothing to be done here.
+
+```js
+new ResizeObserver({}); // TypeError
+```
+
+#### <a id="effects"></a> Effects within the observable elements
+
+Errors due to the resize effect of observable elements and the resize loop (over the boundaries, etc).
+
+Some of these errors may already be recognized by each of Observer API's explainer, like the ResizeObserver's [Deliver Resize Loop error notification](https://drafts.csswg.org/resize-observer-1/#deliver-resize-error).
+
+#### Callback executions
+
+Execution errors when the Observer callback function is called. When unhandled, they bubble out of the callback function execution and eventually hit as an window error.
+
+One way to exemplify this would be an immediate throwing expression:
+
+```js
+new ResizeObserve(function() {
+  throw new Error();
+});
+```
+
+The error from the callback function cannot be captured from registering the elements through the `.observe` method. The registration does not imply a call to that function.
+
+```js
+const observer = new ResizeObserver(function() {
+    throw new Error();
+});
+
+try {
+    observer.observe(elem1);
+} catch (e) {
+    /* the only error to be captured here is if elem1 don't meet the parameters requirements */
+}
+```
+
+The current Observer model has no control over the given error. This error would bubble out and eventually captured in the `window.onerror`. It would be useful for the Observer instances to provide some control to capture and/or handle the error.
```

Comment:
```suggestion
The current Observer model has no control over the given error. This error would bubble out and eventually be captured in the `window.onerror`. It would be useful for the Observer instances to provide some control to capture and/or handle the error.
```